library/src/main/java/com/getstream/sdk/chat/adapter/ChannelListDiffCallback.java

```diff
@@ -53,6 +53,23 @@ public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
         if (!oldChannel.getExtraData().equals(newChannel.getExtraData())) {
             return false;
         }
+        // Check Message Update
+        try {
```

Comment:
This code is terrible; please:

- extract lastMessage once
- do null checks instead of long chains with try/catch